src/Core/Utilities/CustomIpRateLimitMiddleware.cs

```diff
@@ -21,12 +21,12 @@ public class CustomIpRateLimitMiddleware : IpRateLimitMiddleware
             IMemoryCache memoryCache,
             IBlockIpService blockIpService,
             RequestDelegate next,
+            IProcessingStrategy processingStrategy,
+            IRateLimitConfiguration rateLimitConfiguration,
             IOptions<IpRateLimitOptions> options,
-            IRateLimitCounterStore counterStore,
             IIpPolicyStore policyStore,
-            ILogger<CustomIpRateLimitMiddleware> logger,
-            IIpAddressParser ipParser = null)
-            : base(next, options, counterStore, policyStore, logger, ipParser)
+            ILogger<CustomIpRateLimitMiddleware> logger)
+            : base(next, processingStrategy, options, policyStore, rateLimitConfiguration, logger)
```

Comment:
What kind of tests/guarantees do we have that this doesn't change functionality?

Reply:
I have a note in the child task for QA to test the rate limiter but I have also added an integration test that tests the rate limiter blocking a request.